### PR TITLE
feat: Adds Elliptic-Curve Diffie-Helman

### PR DESCRIPTION
**Pull request by HashMapsData2Value**
Adds Elliptic-Curve Diffie-Helman with ECDH().

- [x] Adds tests
- [x] Adds documentation 

(Was broken off from the branch that already has signing tx:ns setup; once that has been merged this PR will not be as big)

### CODE REVIEW

**Comment by ori-shem-tov** [comment]
Can we use a constant for key sizes instead of `32`?

**Comment by HashMapsData2Value** [comment]
> Can we use a constant for key sizes instead of `32`?

Fixed.

File: Tests/bip32-ed25519-swiftTests/TestUtils.swift [hunk 1]
```diff
@@ -1,7 +1,51 @@
...
+ * limitations under the License.
+ */
+
 import Foundation
+import Sodium
+import Clibsodium
 import Base32
 
 public struct TestUtils {
+    public static func cryptoSecretBoxEasy(cleartext: String, nonce: Data, sharedSecret: Data) -> Data {
```
Comment (ori-shem-tov) [inline Tests/bip32-ed25519-swiftTests/TestUtils.swift:24]:
```suggestion
    public static func cryptoSecretBoxEasy(cleartext: String, nonce: Data, secret: Data) -> Data {
```

Comment (ori-shem-tov) [inline Tests/bip32-ed25519-swiftTests/TestUtils.swift:24]: in this context it doesn't have to be a shared secret

File: Tests/bip32-ed25519-swiftTests/bip32-ed25519-swiftTests.swift [hunk 2]
```diff
@@ -19,14 +19,15 @@ import XCTest
 @testable import bip32_ed25519_swift
 import MnemonicSwift
 
-final class Bip32Ed25519Tests: XCTestCase {
+
+final class Bip32Ed25519STests: XCTestCase {
```
Comment (ori-shem-tov) [inline Tests/bip32-ed25519-swiftTests/bip32-ed25519-swiftTests.swift]: why Ed25519`S`?

Comment (HashMapsData2Value) [reply Tests/bip32-ed25519-swiftTests/bip32-ed25519-swiftTests.swift]: Typo that I fixed

File: Tests/bip32-ed25519-swiftTests/TestUtils.swift (hunk 1 above)
Comment (HashMapsData2Value) [reply Tests/bip32-ed25519-swiftTests/TestUtils.swift]: Changed it to symmetricKey (encryption and decryption key are symmetric).

**Review by ori-shem-tov — APPROVED** [review]
LGTM

**Pull request merged**